### PR TITLE
Gce not found issue

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

**Comment by jdoliner** [comment]
@derekchiang whenever you have a chance, it's pretty simple but let me know what you think of the defensive test for IsNotExist, I think it's a good idea but not totally sure, it's a little weird.

File: src/server/pkg/obj/obj.go [hunk 1]
```diff
@@ -314,3 +315,13 @@ func isNetRetryable(err error) bool {
 	netErr, ok := err.(net.Error)
 	return ok && netErr.Temporary()
 }
+
+// TestIsNotExist is a defensive method for checking to make sure IsNotExist is
+// satisfying its semantics.
+func TestIsNotExist(c Client) error {
+	_, err := c.Reader(uuid.NewWithoutDashes(), 0, 0)
```
Comment (derekchiang) [inline src/server/pkg/obj/obj.go]: but are you defensive enough to defend against UUID collision

**Comment by jdoliner** [comment]
I actually think it might make some sense to leave it as something that happens on startup. Even when we have test coverage for object stores we're unlikely to have it for every object store, especially since `minio` can be used with a bunch of different object stores. This seems like a pretty easy way to give a more meaningful error to users rather than the very confusing ones they're bound to get if `NotExist` doesn't work.

File: src/server/pkg/obj/obj.go (hunk 1 above)
Comment (jdoliner) [inline src/server/pkg/obj/obj.go]: Not even close to defensive enough, the good news though is that in order for  this to happen it requires a user not just writing uuid values (without dashes) to the top level of their bucket but writing the correct uuid value, so they need to win the uuid lottery. And if this does happen then the server will just restart and role a new uuid.... if someone has written all the uuids to their bucket I expect pachyderm to have some issues though.

**Pull request merged**